_posts/2018-04-04-workflow-automation.md

```diff
@@ -0,0 +1,182 @@
+liam.addHook({
+    events: [ 'pull_request' ],
+    handler: require('@holidaycheck/liam-tasks/tasks/add-jira-link'),
+    arguments: { githubClient }
+});
+
+function doSomethingInScheduledWay(logger, args) {
+    logger.log(`"foo" value is "${args.foo}"`);
+}
+
+// Run scoped function each second and display value of passed argument
+liam.addCron({
+    time: '* * * * * *',
+    handler: doSomethingInScheduledWay
+    arguments: { foo: 'bar' }
+});
+ 
+// Run liam server on port 3000
+liam.start(3000);
+```
+
+
+
+There's no `laim.json|yaml|conf` file here - that's by intention. Most of the time we were restricted by possibilities of static config, thus we decided to go with pure code approach. Although, one can easily create something like `liam-configurator` which would read mentioned config and call `addHook` and `addCron` respectively. 
+
+We released some general use tasks as part of [liam-tasks](https://www.npmjs.com/package/@holidaycheck/liam-tasks) repository. In addition to these, we're using a couple of internal ones, meant specifically for our architecture, like:
+
+* cleaning deployed branches from staging environment
+
+* keeping our branches in sync with our translation management tool
+
+* posting notifications on Slack on when specific conditions are met (not all of them can be realized by existing integrations)
+
+* reminding people to check whether their Slack's `Working Remotely`  status is still relevant (eg. someone forgot to change status next day) - we have rather open home-office policy
+
+  ![home-office-guard](img/posts/2018-04-04-workflow-automation/home-office-guard.png)
+
+
+You can have one `liam` instance per company, per project, per multiple projects, you can create `liam` instances per specific tasks - it's up to you. For us, one `liam` per project turned out to be a sweet spot, but it doesn't it will work the same for you. 
+
+## Probot
+
+Some time ago GitHub released [probot](https://probot.github.io/) - framework for building GitHub Apps to automate and improve workflow. We're experimenting with this too. For example, Sergii created [issue-prioritizer](https://github.com/holidaycheck/issue-prioritizer), which automatically calculates issue priority based on *Effort*, *Potential* and *Weight* labels: 
+
+![issue-prioritizer](img/posts/2018-04-04-workflow-automation/issue-prioritizer.png)
+
+
+Visible differences in `probot` : 
+
+* it utilizes [Github Apps API](https://developer.github.com/apps/) 
+* it's more opinionated, therefore easier - uses [@octokit/rest.js](https://github.com/octokit/rest.js) as GitHub API library, while in case of `liam` each task can use different client or dependencies
+* API calls are already repository aware (eg. you don't need to specify repository name and owner when you want to make a comment under PR, as this is prepopulated)
+* doesn't support scheduled tasks
```

Comment:
It does support this, but it is a [separate package](https://github.com/probot/scheduler).